azurerm/resource_arm_image.go

```diff
@@ -0,0 +1,417 @@
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"source_virtual_machine_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"os_disk.os_type"},
+			},
+
+			"os_disk": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 1,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+
+						"os_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(compute.Linux),
+								string(compute.Windows),
+							}, true),
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+						},
+
+						"os_state": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(compute.Generalized),
+								string(compute.Specialized),
+							}, true),
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+						},
+
+						"managed_disk_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+							Optional: true,
+						},
+
+						"blob_uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"caching": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  string(compute.None),
+							ValidateFunc: validation.StringInSlice([]string{
+								string(compute.None),
+								string(compute.ReadOnly),
+								string(compute.ReadWrite),
+							}, true),
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+						},
+
+						"size_gb": {
+							Type:     schema.TypeInt,
+							Computed: true,
+							Optional: true,
+						},
+					},
+				},
+			},
+
+			"data_disk": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+
+						"lun": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+
+						"managed_disk_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+							ForceNew: true,
+						},
+
+						"blob_uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"caching": {
+							Type:     schema.TypeString,
+							Optional: true,
+							Default:  "None",
+							ValidateFunc: validation.StringInSlice([]string{
+								string(compute.None),
+								string(compute.ReadOnly),
+								string(compute.ReadWrite),
+							}, true),
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+						},
+
+						"size_gb": {
+							Type:     schema.TypeInt,
+							Optional: true,
```

Comment:
so by Computed I mean we will store the remote value in the state regardless of whether it's specified or not (but it depends if it's retuned from the API) :)